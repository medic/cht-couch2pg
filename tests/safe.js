const safe = require('../libs/medic/safe');

describe('safe', () => {
  const opts = {
    withPasswords: {
      debug: true,
      v4Mode: false,
      couchdbUrl: 'https://simon:vato@rhc-ghana.app.medicmobile.org/medic',
      postgresqlUrl: 'postgres://pilon:palon@localhost:5432/rhc_ghana_upgrade',
      docLimit: 100,
      changesLimit: undefined,
      retryCount: 5,
      sleepMins: 120
    },
    withoutPasswords: {
      couchdbUrl: 'https://rhc-ghana.app.medicmobile.org/medic',
      postgresqlUrl: 'postgres://localhost:5432/rhc_ghana_upgrade',
    }
  };

  it('keeps initial opts and removes passwords from returned obj', () => {
    const optsWithPasswords = opts.withPasswords;
    const safeOpts = safe(optsWithPasswords);
    expect(optsWithPasswords).toEqual(opts.withPasswords);
    expect(safeOpts).toMatchSnapshot();
  });

  it('returns valid couch/pg urls for urls without passwords', () => {
    expect(safe(opts.withoutPasswords)).toMatchSnapshot();
  });

});
