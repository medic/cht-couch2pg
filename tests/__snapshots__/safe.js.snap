// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`safe keeps initial opts and removes passwords from returned obj 1`] = `
Object {
  "changesLimit": undefined,
  "couchdbUrl": "rhc-ghana.app.medicmobile.org/medic",
  "debug": true,
  "docLimit": 100,
  "postgresqlUrl": "localhost:5432/rhc_ghana_upgrade",
  "retryCount": 5,
  "sleepMins": 120,
  "v4Mode": false,
}
`;

exports[`safe returns valid couch/pg urls for urls without passwords 1`] = `
Object {
  "couchdbUrl": "https://rhc-ghana.app.medicmobile.org/medic",
  "postgresqlUrl": "postgres://localhost:5432/rhc_ghana_upgrade",
}
`;
